api/src/main/java/org/openmrs/module/fhirExtension/translators/impl/ObsBasedDiagnosticReportTranslatorImpl.java

```diff
@@ -79,6 +93,24 @@ private void setPresentedForm(DiagnosticReport diagnosticReport, FhirDiagnosticR
 		}
 	}
 	
+	private void setPractitioner(DiagnosticReport diagnosticReport, FhirDiagnosticReport fhirDiagnosticReport) {
+		Set<Provider> performers = fhirDiagnosticReport.getPerformers();
+		if (!performers.isEmpty()) {
+			Provider doctorProvider = performers.iterator().next();
+			Reference reference = practitionerReferenceTranslator.toFhirResource(doctorProvider);
+			diagnosticReport.setPerformer(Collections.singletonList(reference));
+		}
+	}
+	
+	private void setPerformer(DiagnosticReport diagnosticReport, FhirDiagnosticReport fhirDiagnosticReport) {
+		if (!diagnosticReport.getPerformer().isEmpty()) {
+			String reference = diagnosticReport.getPerformer().get(0).getReference();
+			String providerUuid = reference.split("/")[1];
+			Provider doctorProvider = providerService.getProviderByUuid(providerUuid);
+			fhirDiagnosticReport.setPerformers(Collections.singleton(doctorProvider));
```

Comment:
I think this logic would be moved away, when practitionerReferenceTranslator toOpenMrsType function is used

Reply:
removed the above logic and using practitionerReferenceTranslator toOpenMrsType function